packages/superset-ui-query/src/types/Query.ts

```diff
@@ -40,6 +40,11 @@ export type QueryObjectExtras = Partial<{
   where?: string;
 }>;
 
+export type ResidualQueryObjectData = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
```

Comment:
`unknown` might work better than `any` here. https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type